cpp/src/community/detail/common_methods.cuh

```diff
@@ -183,10 +187,67 @@ weight_t compute_modularity(
   return Q;
 }
 
+template <typename map_key_t, typename map_value_t, bool multi_gpu>
+rmm::device_uvector<map_value_t>&& lookup_primitive_values_for_keys(
+  raft::handle_t const& handle,
+  rmm::device_uvector<map_key_t>& map_keys,
+  rmm::device_uvector<map_value_t>& map_values,
+  rmm::device_uvector<map_key_t>& keys_to_lookup)
+{
+  static_assert(std::is_integral<map_key_t>::value);
+  static_assert(std::is_same<map_key_t, std::int32_t>::value ||
+                std::is_same<map_key_t, std::int64_t>::value);
+
+  static_assert(std::is_arithmetic<map_value_t>::value);
+
+  rmm::device_uvector<map_value_t> values_for_sought_keys(0, handle.get_stream());
+
+  if constexpr (multi_gpu) {
+    cugraph::detail::compute_gpu_id_from_ext_vertex_t<map_key_t> vertex_to_gpu_id_op{
+      handle.get_comms().get_size()};
+
+    kv_store_t<map_key_t, map_value_t, false> cluster_key_weight_map(
+      map_keys.begin(),
+      map_keys.end(),
+      map_values.data(),
+      invalid_vertex_id<map_key_t>::value,
+      std::is_floating_point<map_value_t>::value ? std::numeric_limits<map_value_t>::max()
+                                                 : invalid_vertex_id<map_key_t>::value,
```

Comment:
`invalid_vertex_id<map_key_t>::value` is `map_key_t` type and will this going to work say if `map_key_t` is `int32_t` and `map_value_t` is int16_t?